### ISSUE TITLE
Add / remove metric tags transforms

### ISSUE THREAD

**Issue by loony-bean** [timberio/vector#771]
Rel #524

**Adding** static tags will allow ingesting useful meta information, such as env, region, etc.

**Removing** tags will give users control over tagged metric space, and the resources required to store and process it.

Use cases:
- CloudWatch accepts only 10 tags (aka dimensions); user might want to selectively drop unnecessary tags to make sure important tags are not trimmed
- DataDog is limiting multiple unique tag combinations; dropping tags will provide means to control this resource

**Comment by binarylogic** [comment]
Yep, we need to solidify our concepts of tags, fields, and how they work across logs. We brough this up in planning as it is popping up across other work we're doing (#736)

**Comment by lukesteensen** [comment]
This sounds good to me. For now, I think we go with the simple, independent version that just works for metrics. Maybe down the line we can do some work to reduce duplication with `add_fields`, but I wouldn't worry about it for now.
